### ISSUE TITLE
Eigen eigenvalues method throws UnsupportedOperationException

### ISSUE THREAD

**Issue by jeremy-lc** [deeplearning4j/deeplearning4j#5529]
Hello, 

As of issue #927 , the problem does not seem to have been solved, at least within my test. I ran the program under clojure with dependencies : 

```
                 [org.nd4j/nd4j-native-platform "1.0.0-beta" :scope "provided"]
                 [org.deeplearning4j/deeplearning4j-core "1.0.0-beta"]
                 [org.nd4j/nd4j-api "1.0.0-beta"]
```

Setup (creation of the array) : 

```
(ns dev
  (:import [java.util Arrays]
           [org.nd4j.linalg.eigen Eigen]
           [org.nd4j.linalg.factory Nd4j]))

(def double-array-class
  (Class/forName "[D"))

(def ddouble-array-class
  (Class/forName "[[D"))

(defn ddouble-array
  [data]
  (->> (map double-array data)
       (into-array double-array-class)))

(def m
  (ddouble-array [[1 2 3] [2 3 1] [3 1 2]]))

;; Check input
(type m) ;; ==> [[D
(Arrays/deepToString m) ;; ==> [[1.0, 2.0, 3.0], [2.0, 3.0, 1.0], [3.0, 1.0, 2.0]]
```

Then trying : 

```
(Eigen/eigenvalues (Nd4j/create m))
;; ==> UnsupportedOperationException   org.nd4j.linalg.factory.BaseBlasWrapper.geev 
(BaseBlasWrapper.java.463)
```

Seems like the same issue as before

Thanks !

**Comment by saudet** [comment]
That's for you @rcorbish :)

**Comment by rcorbish** [comment]
I noticed that Cuda still does not support non-symmetric matrix eigen decomp. the ge (general) matrix solver does not yet exist. 
Let me take a look to see whether there's anything we can implement directly.
In the meantime would svd work, if you need the eigenvalues only ?

**Comment by saudet** [comment]
Would we need MAGMA or something else like that for CUDA? In any case, just
a CPU-only version would be great for now I think.

**Comment by rcorbish** [comment]
MAGMA doesn't have non-symmetric eigen either, the issue is that the results are complex. Is there a complex array in ND4j? We could return the largest value and it's vector?

**Comment by saudet** [comment]
Ah, complex matrices. So that would be an issue even for LAPACK, right?

**Comment by rcorbish** [comment]
Yes,  that's probably why there's few general eigen solvers in the libraries




Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Comment by agibsonccc** [comment]
Closing — references a removed/rewritten subsystem (DataVec, NLP/Word2Vec, Model Zoo, Spark, UI, or Arbiter). These have been fundamentally rewritten or removed in the upcoming release. Please open a new issue if you experience similar problems with the new version.
